src/fu-util.c

```diff
@@ -1959,16 +1965,120 @@ fu_util_update (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_modify_remote_warning (FuUtilPrivate *priv, FwupdRemote *remote, GError **error)
+{
+#if AS_CHECK_VERSION(0,7,8)
+	AsAgreement *agreement;
+	AsAgreementSection *section;
+	const gchar *tmp;
+	g_autofree gchar *basename = NULL;
+	g_autofree gchar *desc_plain = NULL;
+	g_autofree gchar *fn = NULL;
+	g_autoptr(AsApp) app = NULL;
+	g_autoptr(GHashTable) os_release = NULL;
+	g_autoptr(GString) desc_plain_str = NULL;
+	g_autoptr(SoupURI) uri = NULL;
+
+	/* nothing specified */
+	if (fwupd_remote_get_metainfo_uri (remote) == NULL)
+		return TRUE;
+
+	/* download file to usercache */
+	basename = g_path_get_basename (fwupd_remote_get_metainfo_uri (remote));
+	fn = g_build_filename (g_get_user_cache_dir (), "fwupdmgr", basename, NULL);
+	uri = soup_uri_new (fwupd_remote_get_metainfo_uri (remote));
+	if (!fu_util_download_file (priv, uri, fn, NULL, error))
+		return FALSE;
+
+	/* parse file and get default agreement section */
+	app = as_app_new ();
+	if (!as_app_parse_file (app, fn, AS_APP_PARSE_FLAG_NONE, error))
+		return FALSE;
+	agreement = as_app_get_agreement_by_id (app, "warning");
+	if (agreement == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No agreement found in AppStream file");
+		return FALSE;
+	}
+	section = as_agreement_get_section_default (agreement);
+	if (section == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No default section found in AppStream file");
+		return FALSE;
+	}
+	tmp = as_agreement_section_get_desc (section);
+	if (tmp == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_FOUND,
+				     "No description found in agreement section");
+		return FALSE;
+	}
+
+	/* convert to console text */
+	desc_plain = as_markup_convert_simple (tmp, error);
+	if (desc_plain == NULL)
+		return FALSE;
+
+	/* replace any dynamic values from os-release */
+	desc_plain_str = g_string_new (desc_plain);
+	os_release = fwupd_get_os_release (error);
+	as_utils_string_replace (desc_plain_str, "$OS_RELEASE:NAME$",
+				 g_hash_table_lookup (os_release, "NAME"));
+	as_utils_string_replace (desc_plain_str, "$OS_RELEASE:BUG_REPORT_URL$",
+				 g_hash_table_lookup (os_release, "BUG_REPORT_URL"));
+
+	/* show and ask user to confirm */
+	g_print ("%s\n", desc_plain_str->str);
+	if (!priv->assume_yes) {
+		/* ask for permission */
+		g_print ("\n%s [Y|n]: ",
+			 /* TRANSLATORS: should the remote still be enabled */
+			 _("Enable the remote?"));
+		if (!fu_util_prompt_for_boolean (TRUE)) {
+			g_set_error_literal (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOTHING_TO_DO,
+					     "Declined agreement");
+			return FALSE;
+		}
+	}
+#else
+	if (fwupd_remote_get_metainfo_uri (remote) == NULL)
+		return TRUE;
+	g_print ("Unable to show remote warning as AppStream-glib too old\n");
```

Comment:
It might be worth xdg-opening the metadata URI or at least offering "something" for those with older appstream-glib rather than just ignoring the warning.  Maybe something like:
```
"Enabling this remote will require you to agree to an agreement that the software on your machine is too old to view.  At your convenience, please visit $URL."
"Enable the remote anyway ?"
```

Reply:
Maybe I can use the low-level AsNode interface to get the description that way. I'll try.

Reply:
Fixed.